pkg/cli-runtime/testing/command_table.go

```diff
@@ -251,26 +257,58 @@ func (tc CommandTestCase) Run(t *testing.T, scheme *runtime.Scheme, cmdFactory f
 			},
 		)
 
-		cmd := cmdFactory(ctx, c)
-		cmd.SilenceErrors = true
-		cmd.SilenceUsage = true
-		cmd.SetArgs(tc.Args)
-
 		c.Stdin = bytes.NewBuffer(tc.Stdin)
 		output := &bytes.Buffer{}
-		cmd.SetOutput(output)
 		c.Stdout = output
 		c.Stderr = output
-
 		if tc.ShouldPanic {
 			defer func() {
 				if r := recover(); r == nil {
 					t.Error("expected Execute() to panic")
 				}
 			}()
 		}
+
+		var console *expect.Console
+		var donec chan struct{}
+		if tc.WithConsoleInteractions != nil {
+			if runtime.GOOS == "windows" {
+				t.Skipf("test %q uses pseudo tty not supported yet for windows, skipping tets", tc.Name)
+			}
+			var err error
+			console, err = expect.NewConsole(expect.WithStdout(output), expect.WithDefaultTimeout(300*time.Millisecond))
+			if err != nil {
+				t.Fatalf("failed to create console: %v", err)
+			}
+
+			c.Stdout = console.Tty()
+			c.Stderr = console.Tty()
+			c.Stdin = console.Tty()
+
+			defer console.Close()
+
+			donec = make(chan struct{})
+			go func() {
+				defer close(donec)
+				tc.WithConsoleInteractions(t, console)
+			}()
+		}
+
+		cmd := cmdFactory(ctx, c)
+		cmd.SilenceErrors = true
+		cmd.SilenceUsage = true
+		cmd.SetArgs(tc.Args)
+		cmd.SetOutput(c.Stdout)
+
 		cmdErr := cmd.Execute()
 
+		if console != nil {
```

Comment:
why not close the console in go func? once expecations are checked, call defer on console.Close(). wdyt